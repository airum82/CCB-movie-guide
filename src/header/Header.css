.header {
  background-color: var(--color4);
  height: 13vh;
  margin: 2% 0;
  width: 100%;
}

.header * {
  display: inline-block;
}

.header-nav {
  margin: 2%;
  width: 36%;

}

.header-nav * {
  color: black;
  display: inline-block;
  margin: 2%;
  text-decoration: none;
}

.header-nav *:hover {
  color: white;
}

.header-search {
  display: flex;
  float: right;
  margin: 2%;
  text-align: right;
}

.header-search * {
  height: 36%;
}

.header-search button {
  background-color: var(--color3);
  border: none;
}

.current-category {
  color: white;
}

@media only screen and (max-width: 682px) {
  .header-search,
  .header-nav {
    display: block;
    text-align: center;
  }

  .header-nav {
    margin: 0 auto;
    width: 60%;
  }

  .header-nav * {
    display: inline;
  }

  .nav-spacing {
    margin: 0 10%;
  }

  .header-search {
    display: flex;
    float: none;
    justify-content: center;
    margin: 2% auto;
    width: 100%;
  }
}

@media only screen and (max-width: 495px) {

  .header {
    height: 24vh;
  }
  
  .header-nav {
    padding-top: 2%;
  }

  .header-nav * {
    display: block;
  }

  .header-search * {
    height: auto;
  }

}