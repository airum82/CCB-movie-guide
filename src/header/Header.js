import React from 'react';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ grabSearchTerms, getNewCategory, searchMovies, location, searchTerms }) => {
  return (
    <header className="header">
      <div onClick={getNewCategory} className="header-nav">
        <NavLink 
          to='/now_playing' 
          id="now_playing" 
          className={ location.pathname === '/now_playing' ? 'current-category' : ''}
        >
          Now Playing
        </NavLink>
        <NavLink
          to='/popular' 
          id="popular" 
          className={ location.pathname === '/popular' ? 'current-category nav-spacing' : 'nav-spacing' }
        >
          Popular
        </NavLink>
        <NavLink 
          to='/top_rated' 
          id="top_rated" 
          className={location.pathname === '/top_rated' ? 'current-category' : ''}
        >
          Top Rated
        </NavLink>
      </div>
      <form className="header-search" onSubmit={searchMovies}>
        <input value={searchTerms} type="text" name="search" onChange={grabSearchTerms}/>
        <button>search</button>
      </form>
    </header>
  )
}

Header.propTypes = {
  grabSearchTerms: PropTypes.func,
  getNewCategory: PropTypes.func,
  searchMovies: PropTypes.func,
  location: PropTypes.object,
  searchTerms: PropTypes.string
}

export default Header;