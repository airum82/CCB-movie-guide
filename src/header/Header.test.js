/* eslint-disable */
import React from 'react';
import Header from './Header';
import { shallow } from 'enzyme';

describe('Header', () => {
  let wrapper;
  let props = {
    location: {
      pathname: 'pathname'
    },
    searchTerms: '',
    grabSearchTerms: jest.fn(),
    getNewCategory: jest.fn(),
    searchMovies: jest.fn(),
    searchTerms: 'rob'
  }
  //conditional rendering
  it('renders as expected', () => {
    wrapper = shallow(<Header {...props} />)
    expect(wrapper).toMatchSnapshot();
  })

  it('nowplaying link should have class "currenty-category" when pathname is /now_playing', () => {
    props.location.pathname = '/now_playing';
    wrapper = shallow(<Header {...props} />)
    expect(wrapper.find('#now_playing').hasClass('current-category')).toBe(true);
  })

  it('nowplaying link should not have class "currenty-category" when pathname is not /now_playing', () => {
  props.location.pathname = '/'
  wrapper = shallow(<Header {...props} />)
  expect(wrapper.find('#now_playing').hasClass('current-category')).toBe(false)
  })

  it('popular link should have class "current-category" when pathname is /popular', () => {
  props.location.pathname = '/popular'
  wrapper = shallow(<Header {...props} />)
  expect(wrapper.find('#popular').hasClass('current-category')).toBe(true)
  })

  it('popular link should not have class "current-category" when pathname is not /popular', () => {
    props.location.pathname = '/'
    wrapper = shallow(<Header {...props} />)
    expect(wrapper.find('#popular').hasClass('current-category')).toBe(false)
  })

  it('top_rated link should have class "current-category" when pathname is /top_rated', () => {
  props.location.pathname = '/top_rated'
  wrapper = shallow(<Header {...props} />)
  expect(wrapper.find('#top_rated').hasClass('current-category')).toBe(true)
  })

  it('top_rated link should not have class "current-category" when pathname is not /top_rated', () => {
    props.location.pathname = '/'
    wrapper = shallow(<Header {...props} />)
    expect(wrapper.find('#top_rated').hasClass('current-category')).toBe(false)
  })
  //calling prop functions
  beforeEach(() => {
    wrapper = shallow(<Header {...props} />)
  })

  it('should call getNewCategory when nav is clicked on', () => {
    wrapper.find('.header-nav').simulate('click');
    expect(props.getNewCategory).toHaveBeenCalled();
  })

  it('should call searchMovies when form is clicked on', () => {
    wrapper.find('.header-search').simulate('submit')
    expect(props.searchMovies).toHaveBeenCalled()
  })

  it('should call grabSearchTerms when typing into search input', () => {
    wrapper.find('.header-search input').simulate('change')
    expect(props.grabSearchTerms).toHaveBeenCalled()
  })

})
