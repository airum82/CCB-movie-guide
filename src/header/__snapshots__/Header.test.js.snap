// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected 1`] = `
<header
  className="header"
>
  <div
    className="header-nav"
    onClick={[MockFunction]}
  >
    <NavLink
      activeClassName="active"
      aria-current="page"
      className=""
      id="now_playing"
      to="/now_playing"
    >
      Now Playing
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="nav-spacing"
      id="popular"
      to="/popular"
    >
      Popular
    </NavLink>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className=""
      id="top_rated"
      to="/top_rated"
    >
      Top Rated
    </NavLink>
  </div>
  <form
    className="header-search"
    onSubmit={[MockFunction]}
  >
    <input
      name="search"
      onChange={[MockFunction]}
      type="text"
      value="rob"
    />
    <button>
      search
    </button>
  </form>
</header>
`;
